Extension/package.json

```diff
@@ -282,2546 +282,2587 @@
+          },
+          "C_Cpp.inlayHints.autoDeclarationTypes.showOnLeft": {
+            "type": "boolean",
+            "default": false,
+            "markdownDescription": "%c_cpp.configuration.inlayHints.autoDeclarationTypes.showOnLeft.markdownDescription%",
+            "scope": "resource"
+          },
+          "C_Cpp.inlayHints.parameterNames.enabled": {
+            "type": "boolean",
+            "default": false,
+            "markdownDescription": "%c_cpp.configuration.inlayHints.parameterNames.enabled.markdownDescription%",
+            "scope": "resource"
+          },
+          "C_Cpp.inlayHints.parameterNames.suppressWhenArgumentContainsName": {
+            "type": "boolean",
+            "default": true,
+            "markdownDescription": "%c_cpp.configuration.inlayHints.parameterNames.suppressWhenArgumentContainsName.markdownDescription%",
+            "scope": "resource"
+          },
+          "C_Cpp.inlayHints.parameterNames.hideLeadingUnderscores": {
+            "type": "boolean",
+            "default": true,
+            "markdownDescription": "%c_cpp.configuration.inlayHints.parameterNames.hideLeadingUnderscores.markdownDescription%",
+            "scope": "resource"
+          },
+          "C_Cpp.inlayHints.referenceOperator.enabled": {
+            "type": "boolean",
+            "default": false,
+            "markdownDescription": "%c_cpp.configuration.inlayHints.referenceOperator.enabled.markdownDescription%",
+            "scope": "resource"
+          },
+          "C_Cpp.inlayHints.referenceOperator.showSpace": {
+            "type": "boolean",
+            "default": false,
+            "markdownDescription": "%c_cpp.configuration.inlayHints.referenceOperator.showSpace.markdownDescription%",
+            "scope": "resource"
+          },
+          "C_Cpp.references.maxConcurrentThreads": {
+            "type": [
+              "integer",
+              "null"
+            ],
+            "markdownDescription": "%c_cpp.configuration.references.maxConcurrentThreads.markdownDescription%",
+            "default": null,
+            "minimum": 1,
+            "maximum": 32,
+            "scope": "machine"
+          },
+          "C_Cpp.references.maxCachedProcesses": {
+            "type": [
+              "integer",
+              "null"
+            ],
+            "markdownDescription": "%c_cpp.configuration.references.maxCachedProcesses.markdownDescription%",
+            "default": 0,
+            "minimum": 0,
+            "maximum": 32,
+            "scope": "machine"
+          },
+          "C_Cpp.references.maxMemory": {
+            "type": [
+              "integer",
+              "null"
+            ],
+            "markdownDescription": "%c_cpp.configuration.references.maxMemory.markdownDescription%",
+            "default": null,
+            "minimum": 256,
+            "maximum": 65536,
+            "scope": "machine"
+          },
+          "C_Cpp.files.exclude": {
```

Comment:
This setting theoretically can apply to all features and not just IntelliSense, so it seems like it should be moved out, e.g. it'll get used by Code Analysis too (and that's in a different section), and we have a bug/TODO tracking making it work for formatting.